tests/smoke/smoke_test.go

```diff
@@ -1,13 +1,134 @@
 package smoke_test
 
 import (
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"time"
+
+	"github.com/accuknox/auto-policy-discovery/src/types"
+	"github.com/kubearmor/KubeArmor/tests/util"
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	"github.com/rs/zerolog/log"
 )
 
+func getMatchDir(policy types.KubeArmorPolicy, sys string, str string) string {
+	if sys == "file" {
+		for i := range policy.Spec.File.MatchDirectories {
+			if policy.Spec.File.MatchDirectories[i].Dir == str {
+				value := str
+				return value
+			}
+		}
+	} else {
+		for i := range policy.Spec.Process.MatchDirectories {
+			if policy.Spec.Process.MatchDirectories[i].Dir == str {
+				value := str
+				return value
+			}
+		}
+	}
+	return ""
+}
+
+func getMatchPath(policy types.KubeArmorPolicy, sys string, str string) string {
+	if sys == "file" {
+		for i := range policy.Spec.File.MatchPaths {
+			if policy.Spec.File.MatchPaths[i].Path == str {
+				value := str
+				return value
+			}
+		}
+	} else {
+		for i := range policy.Spec.Process.MatchPaths {
+			if policy.Spec.Process.MatchPaths[i].Path == str {
+				value := str
+				return value
+			}
+		}
+	}
+	return ""
+}
+
+func checkPod(name string, ant string, ns string) {
+	pods, err := util.K8sGetPods(name, ns, []string{ant}, 60)
+	Expect(err).To(BeNil())
+	Expect(len(pods)).To(Equal(1))
+}
+
+var _ = BeforeSuite(func() {
+	// install discovery-engine
+	_, err := util.Kubectl(fmt.Sprintf("apply -f https://raw.githubusercontent.com/kubearmor/discovery-engine/dev/deployments/k8s/deployment.yaml"))
+	Expect(err).To(BeNil())
+	time.Sleep(20 * time.Second)
```

Comment:
This `sleep` here is a risky thing to do.
In Kubearmor, we have a way to check whether the deployment is up by verifying the status of the deployed pods.

Reply:
I think to check the pod is up or not is already implemented [here](https://github.com/accuknox/discovery-engine/pull/653/files#diff-d7590c6b861cadd571109c26d16100a5b65b48829ede89e2eb611126004fc701R48-R52)

Reply:
In kubearmor we verify whether the deployment is up by checking if the relevant pod is in Running condition.